python/api/config/views.py

```diff
@@ -0,0 +1,147 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from django.db import models
+from rest_framework.response import Response
+
+from api.api_views import ListView, GenericAPIPermView, create, update
+from api.serializers import check_obj, get_config_version
+from cm.adcm_config import ui_config
+from cm.errors import AdcmApiEx
+from cm.models import ADCM, Cluster, HostProvider, Host, ClusterObject, ConfigLog, ObjectConfig
+from . import serializers
+
+
+def get_objects_for_config(object_type):
+    if object_type == 'adcm':
+        return ADCM.objects.all()
+    if object_type == 'cluster':
+        return Cluster.objects.all()
+    elif object_type == 'provider':
+        return HostProvider.objects.all()
+    elif object_type == 'service':
+        return ClusterObject.objects.all()
+    elif object_type == 'host':
+        return Host.objects.all()
+    else:
+        # This function should return a QuerySet, this is necessary for the correct
+        # construction of the schema.
+        return Cluster.objects.all()
+
+
+def get_obj(objects, object_type, object_id):
+    try:
+        obj = objects.get(id=object_id)
+    except models.ObjectDoesNotExist:
+        errors = {
+            'adcm': 'ADCM_NOT_FOUND',
+            'cluster': 'CLUSTER_NOT_FOUND',
+            'provider': 'PROVIDER_NOT_FOUND',
+            'host': 'HOST_NOT_FOUND',
+            'service': 'SERVICE_NOT_FOUND',
+        }
+        raise AdcmApiEx(errors[object_type]) from None
+
+    if object_type == 'provider':
+        object_type = 'hostprovider'
+    if object_type == 'service':
+        object_type = 'clusterobject'
+    oc = check_obj(ObjectConfig, {object_type: obj}, 'CONFIG_NOT_FOUND')
+    cl = ConfigLog.objects.get(obj_ref=oc, id=oc.current)
+    return obj, oc, cl
+
+
+class ConfigView(ListView):
+    serializer_class = serializers.HistoryCurrentPreviousConfigSerializer
+    object_type = None
+
+    def get_queryset(self):
+        return get_objects_for_config(self.object_type)
+
+    def get(self, request, *args, **kwargs):
+        object_type = kwargs['object_type']
+        self.object_type = object_type
+        object_id = kwargs[f'{object_type}_id']
```

Comment:
This 3 string repeated too much. I suggest to write a function

object_type, object_id = get_obj_type_and_id(kwargs)

Reply:
Fixed in https://github.com/arenadata/adcm/pull/563/commits/2ff1cbcf6ee5573650cd2fdf0928c4ac22360223